Fixes #372

adamrp opened the pull request:
Before attempting to send mail, the hostname is retrieved using
socket.gethostname. If the hostname includes 'microbio.me', then mail is sent.
Otherwise, the user is dumped back to the portal with a message stating that
mail can be sent only from the microbio.me domain.

Commit:
Fixes #372

Before attempting to send mail, the hostname is retrieved using
socket.gethostname. If the hostname includes 'microbio.me', then mail is sent.
Otherwise, the user is dumped back to the portal with a message stating that
mail can be sent only from the microbio.me domain.

ElDeveloper commented on www/americangut/add_participant.psp:
```diff
@@ -13,7 +13,7 @@ __status__ = "Development"
 from enums import ServerConfig,DataAccessType
 from utils.mail import send_email
-import socket
+from socket import gethostname
```
Would it make more sense to have a function `is_live_system` or `can_send_email` that would return a `True` or `False`? Then all the `if 'microbe.me' in hostname` statements are factored into this single function.

Commit:
Merge branch 'master' of https://github.com/qiime/qiime_web_app into i372

Commit:
Add function can_send_mail

takes no arguments, returns true if 'microbio.me' is in socket.hostname();
returns false otherwise

Commit:
Adding doc string and streamlined fn

Commit:
Modify files to use can_send_mail

Files should use can_send_mail from utils.mail instead of explicitly checking
whether or not 'microbio.me' appears in socket.hostname() return value

douginator2000 commented:
Are we confident that email functionality is still good?

adamrp commented:
I tested on webdev, and it seems to work fine.  I recommend testing though, to reassure yourself.

Commit:
Merge branch 'master' of https://github.com/qiime/qiime_web_app into i372

Conflicts:
	www/americangut/submit_human_survey.psp

Commit:
Add better error catching on add_participant.psp
